samples/instanceadmin/test_instanceadmin.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import random
+
+import instanceadmin
+
+
+PROJECT = os.environ["GOOGLE_CLOUD_PROJECT"]
+ID_FORMAT = "instanceadmin-test-{:06}"
+ID_RANGE = 1000000
+
+INSTANCE = ID_FORMAT.format(random.randrange(ID_RANGE))
+CLUSTER1 = ID_FORMAT.format(random.randrange(ID_RANGE))
+CLUSTER2 = ID_FORMAT.format(random.randrange(ID_RANGE))
+
+
+def test_run_instance_operations(capsys):
+    try:
+        instanceadmin.run_instance_operations(PROJECT, INSTANCE, CLUSTER1)
+        out = capsys.readouterr().out
+        assert f"Instance {INSTANCE} does not exist." in out
+        assert "Creating an instance" in out
+        assert f"Created instance: {INSTANCE}" in out
+        assert "Listing instances" in out
+        assert f"\n{INSTANCE}\n" in out
+        assert f"Name of instance: {INSTANCE}" in out
+        assert "Labels: {'prod-label': 'prod-label'}" in out
+        assert "Listing clusters..." in out
+        assert f"\n{CLUSTER1}\n" in out
+
+        instanceadmin.run_instance_operations(PROJECT, INSTANCE, CLUSTER1)
+        out = capsys.readouterr().out
+        assert f"Instance {INSTANCE} already exists." in out
+        assert "Listing instances" in out
+        assert f"\n{INSTANCE}\n" in out
+        assert f"Name of instance: {INSTANCE}" in out
+        assert "Labels: {'prod-label': 'prod-label'}" in out
+        assert "Listing clusters..." in out
+        assert f"\n{CLUSTER1}\n" in out
+
+    finally:
+        instanceadmin.delete_instance(PROJECT, INSTANCE)
```

Comment:
is there a way to try again if it fails or just make sure it gets deleted? If too many tests fail the deletion the project will run out of Bigtable nodes and fail and they will need to be cleaned up manually. Maybe a check at start of test for instances starting with instanceadmin and deleting? not sure the best approach

Reply:
8b30d5c

The trade off is now we can screw up another test run or if there is more than one test run happening in parallel for the same project. I don't really see that coming up in practice, though.

Reply:
Looks like the preclean is the wrong approach, because of the race condition it introduces it. I guessed we'd be unlikely to hit it, but guessed wrong. I understand your concern here and can have it try again a few times of a deletion fails, but ultimately, a few might get through and have to be removed manually.

Reply:
Usually we would wrap the "must succeed" teardown stuff in a retry, as in [`docs/snippets.py`](https://github.com/googleapis/python-bigtable/blob/e001812cea997ffb56811195b2d59a8d4266fadc/docs/snippets.py#L101-L108).

Reply:
Oh, that's great. I hadn't seen that before. In this case, since the failure is hypothetical, I'm not sure which exceptions should be trapped here, as I've never seen a failure.